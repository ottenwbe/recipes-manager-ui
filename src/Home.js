import React from 'react';
import Paper from '@mui/material/Paper';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardContent from '@mui/material/CardContent';
import { PageHeader } from './PageHeader';
import { Link } from 'react-router-dom';
import { textFromContext } from './common/context/TextContextProvider';

export function Home(props) {

        const texts = useContext(TextContext);

        return (<React.Fragment>
            <PageHeader pageName="Recipe News" />
            <Card style={{ align: 'center' }}>
                <CardHeader title={textFromContext(texts, 'welcome')} subheader={textFromContext(texts, 'welcomeSub')} style={{ textAlign: 'center', backgroundColor: "#2196f3" }} />
                <CardContent>
                    <Paper />

                        Look at all our recipes at:

                        <p>
                            <Link to="/recipes">Recipes</Link>
                        </p>

                        Or get a random Recipe at:

                        <p>
                            <Link to="/rand">Random Recipes</Link>
                        </p>
                </CardContent>
            </Card>
        </React.Fragment>);
    
}
