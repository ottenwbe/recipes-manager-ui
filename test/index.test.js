import { render } from '@testing-library/react'
import Home from '../src/pages/index'

it('renders homepage unchanged', () => {
  //TODO: fails due to fetch problem
  //const { container } = render(<Home />)
  //expect(container).toMatchSnapshot()
})