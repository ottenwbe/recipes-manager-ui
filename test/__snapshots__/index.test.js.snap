// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    style="text-align: center;"
  >
    Loading...
  </div>
  <div
    class="Footer"
    style="text-align: center;"
  >
    <hr
      style="color: black; background-color: black; height: 1px; width: 100px;"
    />
    <p />
  </div>
</div>
`;
